addons/l10n_in/tests/test_partner_details_on_invoice.py

```diff
@@ -139,3 +139,41 @@ def test_partner_change_with_invoice(self):
                 'l10n_in_state_id': self.env.company.state_id.id,
             }]
         )
+
+    def test_place_of_supply(self):
+        invoice = self.init_invoice(
+            move_type='out_invoice',
+            partner=self.partner_b,
+        )
+
+        child_partner = self.env['res.partner'].create({
+            'name': "Child Contact",
+            'type': "delivery",
+            'parent_id': self.partner_b.id,
+            'state_id': self.env.ref("base.state_in_gj").id
+        })
+
+        self.assertRecordValues(
+            invoice,
+            [{
+                'partner_shipping_id': self.partner_b.id,
+                'l10n_in_state_id': self.partner_b.state_id.id,
+            }]
+        )
+        invoice.partner_id = self.partner_b.id
+        self.assertRecordValues(
+            invoice,
+            [{
+                'partner_shipping_id': child_partner.id,
+                'l10n_in_state_id': child_partner.state_id.id,
+            }]
+        )
+        child_partner.state_id = False
+        invoice.partner_id = self.partner_b.id
```

Comment:
```suggestion
        invoice.partner_shipping_id = self.partner_a
```
Rather than making child_partner state False I would prefer to check it with a third party partner
May be you can add a bit extra lines or make changes here up to you @masm-odoo